### ISSUE TITLE
FlightAware Pro Stick Problems

### ISSUE THREAD

**Issue by willfrd** [jprochazka/adsb-receiver#304]
Recntly I ubgraded my DVB-T usb to a FA Pro Stick and I am having serious performance problems.
With DVB-T:
![dvb-t](https://cloud.githubusercontent.com/assets/20045335/22175592/bf46bf14-dfef-11e6-854b-c8ce0be3e406.jpg)

FA ProStick:
![fa prostick](https://cloud.githubusercontent.com/assets/20045335/22175595/da34f818-dfef-11e6-915e-1294af8869fa.jpg)

This really surprises me since they sell the ProStick as "20-100% more ADS-B range". I have been trying to use different gain values but there hasn't improved anything. Is this beacuse the software is not compatible with it? Am I doing something wrong?

Thanks

**Comment by jthawkes** [comment]
Check the RSSI level to see if the dongle is being over driven. 
Are you using a filter? The amplifier in the ProStick can be over driven without one.
When I tested a ProStick, I saw 200 aircraft with a filter(Cavity not FA) and 2 without one.
I am in New York City and have a lot of cell/mobile phone towers and radars near me.

Remember that dump1090 needs to be restarted after the changes to the gain settings are made.  You should see changes each time you change the gain. Check the FA forums for gain testing scripts.

The software is compatible.

**Comment by willfrd** [comment]
Thanks for your answer @jthawkes , I don't use a filter. The graphs show aparently what you said the dongle is being overdriven, (the curvy segment is with the DVB-T and the straight part is with the prostick): 
![image](https://cloud.githubusercontent.com/assets/20045335/22178041/53fd871c-e024-11e6-900b-ec490a4760f2.png)
I have the gain set at 50. Is there a solution for this? Can it be fixed by setting a lower gain?

Thanks again.

**Comment by jthawkes** [comment]
from http://flightaware.com/adsb/prostick/
Due to the high amount of RF amplification, it is necessary to install a 1090MHZ filter and/or reduce gain on your receiver (e.g., PiAware/dump1090). Read more about setting up your FlightAware Pro Stick for optimal reception. http://flightaware.com/adsb/prostick/setup/
Noise level is close to signal level.  Try dropping the gain the 30 then increasing it slowly.

**Comment by willfrd** [comment]
In this gragph you can see the the signal level with gain at 0.0, 0.9, 10.0, and 30.0:
![image](https://cloud.githubusercontent.com/assets/20045335/22182317/7eddfa4a-e09a-11e6-9265-49a183dedee7.png)

Always by setting the gain:
`sudo sudo piaware-config rtlsdr-gain <gain>`
Restartingthe piaware:
`sudo systemctl restart piaware`
Restarting the raspberry pi:
`sudo reboot`

Am I following all the steps I should?

**Comment by willfrd** [comment]
Its worth saying that with the dongle came this note:
![img_20170122_185422](https://cloud.githubusercontent.com/assets/20045335/22184966/52a4da4a-e0d4-11e6-86c2-9ad68b1d2fd0.jpg)

With the steps I followed in the website didn't work.

**Comment by ForestFeather** [comment]
There are config files for dump1090 and piaware, that seem to override any rtl settings.  I had to manually specify the gain in my file '/etc/default/dump1090-mutability' before my gain settings took effect.

To figure out your best gain, you may want to look at the forums for piaware.  There are a few scripts that you can run that will work their way through all gain settings and help you find the best gain.  I would recommend running them however during a time you know there is high flight traffic overhead.

**Comment by willfrd** [comment]
Ok, so in order for the gai effects to take place I have to open the /etc/default/dump1090-mutability file and where it says:
```
# RTLSDR gain in dB.
# If set to "max" (the default) the maximum supported gain is used.
# If set to "agc", the tuner AGC is used to set the gain.
GAIN="max"
```
Put the gain I want e.g. :
`GAIN="<gain value e.g. 30>"`

And then do all the steps I deatailed before with the respective gain value.

About the script would it be too much to ask you for a link with the script? (I can't manage to find it in the FA forum)

Thanks!

**Comment by ForestFeather** [comment]
That would get you to run at the correct gain!  To give you an idea, I have the FlightAware stick (not the pro version, however) and dropping my gain from max to 33.8 resulted in tripling my reception rates.

The relevant thread is this: [http://discussions.flightaware.com/ads-b-flight-tracking-f21/gain-adjustment-t37172.html](http://discussions.flightaware.com/ads-b-flight-tracking-f21/gain-adjustment-t37172.html), but you will want to read through the whole thing - there's a few different versions in there.

**Comment by Romeo-Golf** [comment]
Hi, thanks to everyone for their assistance on this one and as highlighted the default gain values on dump1090 are perhaps not ideal when using an LNA, either external or contained within FlightAware's ProStick dongles...

I have created the following issue to track the changes required to support manually overriding the default gain value for each dump1090 fork, which as discussed above does require slightly different steps between the Mutability and FlightAware forks:

https://github.com/jprochazka/adsb-receiver/issues/303

We are also looking to add a graph of the gain value which is configured in dump1090 to assist with tracking the results of these changes:

https://github.com/jprochazka/adsb-receiver/issues/297

And although the scripts mentioned on the FlightAware thread do look interesting, these results could be considered indicative at best, so although the results are probably a good starting point, it doesn’t seem a priority compared to adding new features...  but if this is something that these is a strong demand for then I'm sure that its something which can be considered ;)

**Comment by chrispt** [comment]
So I just discovered this forum thread and glad I did! I had been playing with my gain settings in piaware-config but had never changed them in my dump1090-mutability file. In piaware-config it was set to 50 and in dump1090-mutability it was set to 20 something I think! I have since corrected it and it seems to be better so far. I'll let it run during the day tomorrow. I ran the optimize script on the forum mentioned and it was picking up more planes at main gain where I am located.
